doc/api/esm.md

```diff
@@ -431,6 +431,46 @@ thrown:
 }
 ```
 
+#### Self-referencing a package using its name
+
+Self-referencing a package's name in an `import` is allowed. For example,
+let's assume the `package.json` is:
+
+```json
+// package.json
+{
+  "name": "a-package",
+  "exports": {
+    ".": "./main.mjs",
+    "/cjs": "./main.js"
```

Comment:
```suggestion
    "./cjs": "./main.js"
```

Reply:
I think having the one example for this involve two exports that are really the same code, just alternate ESM/CommonJS variants of it, confuses the purpose. Can we just leave dual-mode out of this, and make the exports be something like `".": "./main.mjs"` and "`./foo": "./foo.mjs"` or something?

Reply:
I don't think that would work with the example below that uses `require`. We could make the import target  consistently CJS though.

Reply:
I think consistently CommonJS is good. 👍